internal-ui/src/components/DashboardComponent.tsx

```diff
@@ -1,23 +1,76 @@
-import React from 'react';
-import { CalendarMonth } from '@patternfly/react-core';
-import { Checkbox } from '@patternfly/react-core';
+import React, { useEffect, useState } from 'react';
 import {
+  CalendarMonth,
+  Checkbox,
   Card,
-  CardBody,
-  CardTitle,
-  CardHeader,
+  Grid,
+  GridItem,
 } from '@patternfly/react-core';
-import { useState } from 'react';
-
+import {
+  ChartDonut,
+  ChartPie,
+  ChartArea,
+  ChartScatter,
+  ChartThemeColor,
+  ChartVoronoiContainer,
+  ChartBar
+} from '@patternfly/react-charts';
+import { getMessagesInRange } from '../services/messages';
+import { Message } from '../Types';
 
-export const SelectFilters = () => {
+export const DashboardComponent = () => {
   const [startDate, setStartDate] = useState<Date | null>(null);
   const [endDate, setEndDate] = useState<Date | null>(null);
+  const [messages, setMessages] = useState<Message[]>([]);
+  const [uniqueSessionsCount, setUniqueSessionsCount] = useState(10);
+  const [uniqueSenders, setUniqueSenders] =useState(10);
+  const [totalSessions, setTotalSessions] = useState(60);
+  const [averageFlows, setAverageFlows] = useState(1000);
+  const [thumbsUp, setThumbsUpCount] = useState(50);
+  const [thumbsDown, setThumbsDownCount] = useState(40);
+  const [toggleState, setToggleState] = useState({
+    showAll: true,  
+    showInternal: false,
+    showExternal: false
+  });
+
+  useEffect(() => {
+    if (startDate && endDate) {
+      const fetchMessages = async () => {
+        const sessionMessages = await getMessagesInRange(undefined, Math.floor(startDate.getTime() / 1000), Math.floor(endDate.getTime() / 1000));
+        setMessages(sessionMessages);
+      };
+      fetchMessages();
+    }
+    setTotalSessions(messages.length)
+    
+    const uniqueSessions =new Set(messages.map(msg => msg.id)).size;
+    setUniqueSessionsCount(uniqueSessions);
+
+    const uniqueSenders = new Set(messages.map(msg =>msg.sender_id)).size;
+    setUniqueSenders(uniqueSenders);
+
+    function isSlotMessage(msg: Message): boolean {
+      return msg.type_name === 'slot' && msg.data.name === 'closing_feedback';
+    }
+    
+    const thumbsUp = messages.filter(msg => isSlotMessage(msg)).length;
+    const thumbsDown = messages.filter(msg => !isSlotMessage(msg)).length; 
+    setThumbsUpCount(thumbsUp); 
+    setThumbsDownCount(thumbsDown);
+ 
+
+    const totalConversations = messages.reduce((acc,msg) => acc + msg.id, 0);
+    const averageConversations = messages.length > 0 ? (totalConversations / messages.length) : 0;
+    setAverageFlows(averageConversations);
+    setTotalSessions(messages.length)
 
-  const handleDateChange = (_event: React.MouseEvent<HTMLButtonElement, MouseEvent>, date: Date) => {
+  }, [startDate, endDate,messages]);
```

Comment:
Simiarly, I can't tell where the start/end date come from, but if the end date is the current day, we would want it to be the current time. If it's a previous day, the time should be set to the end of that day.